src/PiGui.cpp

```diff
@@ -0,0 +1,241 @@
+		Error("Could not alloc image buffer.\n");
+	}
+	{
+		float scale = double(W)/w;
+		float tx = 0;
+		float ty = 0;
+		nsvgRasterize(rast, image, tx, ty, scale, img, W, H, W*4);
+	}
+	nsvgDeleteRasterizer(rast);
+	nsvgDelete(image);
+	pigui->icons = pigui->makeTexture(img, W, H);
+}
+
+void PiGui::Init(SDL_Window *window) {
+	m_handlers.Unref();
+
+	lua_State *l = Lua::manager->GetLuaState();
+	lua_newtable(l);
+	m_handlers = LuaRef(l, -1);
+
+	lua_newtable(l);
+	m_keys = LuaRef(l, -1);
+	LuaTable keys(l, -1);
+	for(auto p : keycodes) {
+		keys.Set(p.first, p.second);
+	}
+
+	InitIcons(this);
+
+	ImGui_ImplSdlGL3_Init(window);
+
+	ImGuiIO &io = ImGui::GetIO();
+	static unsigned short glyph_ranges[] = { 0x1, 0x3c0, 0x0, 0x0 };
+	pionillium12 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "PionilliumText22L-Medium.ttf").c_str(), 12.0f, nullptr, glyph_ranges);
+	pionillium15 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "PionilliumText22L-Medium.ttf").c_str(), 15.0f, nullptr, glyph_ranges);
+	pionillium18 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "PionilliumText22L-Medium.ttf").c_str(), 18.0f, nullptr, glyph_ranges);
+	pionillium30 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "PionilliumText22L-Medium.ttf").c_str(), 30.0f, nullptr, glyph_ranges);
+	pionillium36 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "PionilliumText22L-Medium.ttf").c_str(), 36.0f, nullptr, glyph_ranges);
+	orbiteer18 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "Orbiteer-Bold.ttf").c_str(), 18.0f, nullptr, glyph_ranges);
+	orbiteer30 = io.Fonts->AddFontFromFileTTF(FileSystem::JoinPath(FileSystem::JoinPath(FileSystem::GetDataDir(), "fonts"), "Orbiteer-Bold.ttf").c_str(), 30.0f, nullptr, glyph_ranges);
+}
+
+int PiGui::RadialPopupSelectMenu(const ImVec2& center, std::string popup_id, std::vector<ImTextureID> tex_ids, std::vector<std::pair<ImVec2,ImVec2>> uvs, unsigned int size, std::vector<std::string> tooltips)
+{
+	int ret = -1;
+
+	// FIXME: Missing a call to query if Popup is open so we can move the PushStyleColor inside the BeginPopupBlock (e.g. IsPopupOpen() in imgui.cpp)
+	// FIXME: Our PathFill function only handle convex polygons, so we can't have items spanning an arc too large else inner concave edge artifact is too visible, hence the ImMax(7,items_count)
+	ImGui::PushStyleColor(ImGuiCol_WindowBg, ImVec4(0,0,0,0));
+	ImGui::PushStyleColor(ImGuiCol_PopupBg, ImVec4(0,0,0,0));
+	ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0,0,0,0));
+	if (ImGui::BeginPopup(popup_id.c_str())) {
+			const ImVec2 drag_delta = ImVec2(ImGui::GetIO().MousePos.x - center.x, ImGui::GetIO().MousePos.y - center.y);
```

Comment:
Double indentation here?

Please sqush the changes into this commit, and update.